test/lit/passes/simplify-globals-offsets.wast

```diff
@@ -0,0 +1,78 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; NOTE: This test was ported using port_passes_tests_to_lit.py and could be cleaned up.
+
+;; RUN: foreach %s %t wasm-opt --simplify-globals -all -S -o - | filecheck %s
+
+;; Apply constant globals to segment offsets. The non-imported global.gets will
+;; be applied in the segments named $use-defined.
+
+(module
+  ;; CHECK:      (type $0 (func))
+
+  ;; CHECK:      (import "env" "memory" (memory $memory 256))
+  (import "env" "memory" (memory $memory 256))
+
+  ;; CHECK:      (import "env" "table" (table $table 0 funcref))
+  (import "env" "table" (table $table 0 funcref))
+
+  ;; CHECK:      (import "env" "imported" (global $imported i32))
+  (import "env" "imported" (global $imported i32))
+
+  ;; CHECK:      (global $defined i32 (i32.const 42))
+  (global $defined i32 (i32.const 42))
+
+  ;; CHECK:      (global $use-defined i32 (i32.const 42))
+  (global $use-defined i32 (global.get $defined))
+
+  ;; CHECK:      (data $use-imported (global.get $imported) "hello, world!")
+  (data $use-imported (global.get $imported) "hello, world!")
+
+  ;; CHECK:      (data $use-defined (i32.const 42) "hello, world!")
+  (data $use-defined (global.get $defined) "hello, world!")
+
+  ;; CHECK:      (data $dropped "hello, world!")
+  (data $dropped "hello, world!")
```

Comment:
Is this relevant to this PR?

Reply:
It's a segment without an offset, so the pass has nothing to do there but also it should not error on it (there is a nullptr for the offset). I added a comment to clarify.